krm-functions/nad-fn/README.md

```diff
@@ -0,0 +1,116 @@
+# NAD function Design
+
+Purpose of the function
+-----------------------
+
+The `NAD` KRM function is designed to be a part of the pipeline of a Nephio NF kpt package.
+
+Its primary purpose is to find all `Network Function Deployments` custom resources and the `IP allocations` in the package and expand it to a set of network attachement definition resources, i.e. `NetworkAttachmentDefinition` then to be used by container network interface (CNI) plugin e.g. Multus. 
+
+The function is also designed to be part of the "`Condition` choreography" that is meant to synchronize the effects of multiple KRM functions acting on the same kpt package. It adds `Conditions` to the `Status` of the `Kptfile` object indicating that the `NetworkAttachmentDefinition` requests are not fulfilled yet. 
+
+Child resources generated for an `NAD` resource
+-------------------------------------------------------
+
+For each `NAD` the function will generate one `NetworkAttachmentDefinition` CR. In order to generate the `Spec` of this child resource it uses information also from the `ClusterContext`, `Network Function Deployments`, `IPAllocation` resource that the kpt package assumed to also contain. 
```

Comment:
/s/resource/resources

Reply:
Can you please detail this comment? Couldn't follow.

Reply:
sorry , I meant to say replace "resource" in the second line  to "resources".